File: CX/ClientApp/src/servicios/ServicioCanales.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "canales";

export const ObtenerCanales = async () => {
    const url = `${controlador}/obtenercanales`;
    return await ProcesarDatosApi('GET', url);
}

export const InsertarCanales = async (data) => {
    const url = `${controlador}/insertarcanales`;
    return await ProcesarDatosApi('POST', url, data);
}
export const ActualizarCanales = async (data) => {
    const url = `${controlador}/actualizarcanales`;
    return await ProcesarDatosApi('PUT', url, data);
}
export const InactivarCanales = async (id) => {
    const url = `${controlador}/eliminarcanales?idCanal=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}

export const ObtenerCanalesPorID = async (id) => {
    const url = `${controlador}/ObtenerCanalesPorID/${id}`;
    return await ProcesarDatosApi('GET', url);
}

File: BussinesLogic/Servicios/ServicioCanales.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using System;
using System.Collections.Generic;
using System.Text;
using Dominio.Interfaces.Aplicacion.Canales;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioCanales : IServicioCanales
    {
        private readonly IRepositorioCanales repositorioCanales;

        public ServicioCanales(IRepositorioCanales repositorioCanales)
        {
            this.repositorioCanales = repositorioCanales;
        }

        public async Task<List<DtoCanales>> ObtenerCanales()
        {
            return await this.repositorioCanales.ObtenerCanales();
        }

        public async Task<DtoRespuestaSP> InsertarCanales(EntitiCanales entitiCanales)
        {
            return await this.repositorioCanales.InsertarCanales(entitiCanales);
        }
        public async Task<DtoRespuestaSP> ActualizarCanales(EntitiCanales entitiCanales)
        {
            return await this.repositorioCanales.ActualizarCanales(entitiCanales);
        }
        public async Task<DtoRespuestaSP> EliminarCanales(int idCanal)
        {
            return await this.repositorioCanales.EliminarCanales(idCanal);
        }
        public async Task<DtoCanales> ObtenerCanalesPorID(int idCanal)
        {
            return await this.repositorioCanales.ObtenerCanalesPorID(idCanal);
        }
    }
}

